### ISSUE TITLE
The Select recording directory option in recorder docker opens file selector instead of folder selector

### ISSUE THREAD

**Issue by raghu-kamath** [flathub/org.kde.krita#50]
In Krita 5.0 there is a new feature of recording timelapse. In the options for this feature the user can select where to save the recording of the paintings. But when the user clicks on the folder icon to browse and select a folder, the flatpak Krita opens the file selector and it is not possible to choose a folder. 

To check the issue, first open the recorder docker from "Settings > Dockers > Recorder" A docker will open in the UI somewhere. Go to the recorder docker and click on the folder icon besides the path in the recording directory. try to choose a folder.
![Screenshot_20220121_140219-1](https://user-images.githubusercontent.com/93243082/150493476-02ede9bc-ebd9-4a5d-8c88-954cdb77bd75.jpeg)

**Comment by vulongm** [comment]
@travier Any ideas how to go about solving this bug? Not sure if we need to patch something with Qt or Krita.

**Comment by travier** [comment]
Have you tried opening the current folder with `.`?

This is only a workaround however and a real fix might involve changes in the portal or Qt.

**Comment by grulja** [comment]
> Have you tried opening the current folder with `.`?
> 
> This is only a workaround however and a real fix might involve changes in the portal or Qt.

I will look into this.

**Comment by travier** [comment]
You should be able to use the current folder by entering `.` in the name field.

**Comment by vulongm** [comment]
> You should be able to use the current folder by entering `.` in the name field.

Trying that the name field clears out when pressing open. The directory Krita uses seems to be the same as the default.

**Comment by grulja** [comment]
Yeah, it's a bug in Krita from Flathub. It bundles quite an old version of qtbase and therefore the portal support there is incomplete. 

For reference this qtbase change is missing: https://github.com/qt/qtbase/commit/844967d297327dc72a1aa67644c1b2fa3fe6839c.

**Comment by vulongm** [comment]
Think it's something that would be possible to patch in the older qtbase? Might be able to try rebasing on the KDE 5.15-21.08 runtime as well but that might result in other bugs.

**Comment by vulongm** [comment]
@grulja Looks like the build with just that patch is failing. Not sure if we're missing some other things. 
```
========================================================================
Building module qtbase in /srv/buildbot/worker/build-x86_64-4/build/.flatpak-builder/build/qtbase-1
========================================================================
]2;flatpak-builder: Building qtbaseFB: Running: tar xf /srv/buildbot/worker/downloads/downloads/7c15c31590b6ec12408242b1a32075d1d545f35e1d41fdbad31ec670d00680f2/qtbase-everywhere-src-5.12.12.tar.xz --no-same-owner --strip-components=1 -J
Applying patch portal.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-4/build/patches/qt/base/portal.patch
patching file src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportalfiledialog.cpp
patching file src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportaltheme.cpp
Applying patch FileChooser-portal-skip-empty-filters.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-4/build/patches/qt/base/FileChooser-portal-skip-empty-filters.patch
patching file src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportalfiledialog.cpp
Hunk #1 succeeded at 232 (offset 5 lines).
Applying patch qtbase-avoid-hardcoding-kernel-version.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-4/build/patches/qt/base/qtbase-avoid-hardcoding-kernel-version.patch
patching file src/corelib/configure.json
Hunk #1 succeeded at 595 (offset 94 lines).
Hunk #2 succeeded at 729 (offset 100 lines).
Hunk #3 succeeded at 739 (offset 100 lines).
Applying patch qtbase-use-wayland-on-gnome.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-4/build/patches/qt/base/qtbase-use-wayland-on-gnome.patch
patching file src/gui/kernel/qguiapplication.cpp
Hunk #1 succeeded at 1320 (offset -56 lines).
Applying patch qtbase-make-sure-to-correctly-construct-base-platform-theme.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-4/build/patches/qt/base/qtbase-make-sure-to-correctly-construct-base-platform-theme.patch
patching file src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportaltheme.cpp
Hunk #1 FAILED at 59.
1 out of 1 hunk FAILED -- saving rejects to file src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportaltheme.cpp.rej
]2;flatpak-builder: Cleanup qtbaseError: module qtbase: Child process exited with code 1
FB: Unmounting read-only fs: fusermount -uz /srv/buildbot/worker/build-x86_64-4/build/.flatpak-builder/rofiles/rofiles-kFltX7
dbus-daemon[2026814]: Reloaded configuration
command timed out: 3600 seconds without output running [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'x86_64', b'--bundle-sources', b'--mirror-screenshots-url=https://dl.flathub.org/repo/screenshots', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-x86_64-4/../downloads', b'--default-branch', b'test', b'--subject', b'Add portal patch to manifest (3a04b110)', b'--remove-tag=upstream-maintained', b'builddir', b'org.kde.krita.yaml'], attempting to kill
program finished with exit code 1
elapsedTime=3630.608441
```

**Comment by grulja** [comment]
It needs to be rebased, apparently it doesn't apply clearly.

**Comment by vulongm** [comment]
> It needs to be rebased, apparently it doesn't apply clearly.

Do you have suggestions to go about that?

**Comment by grulja** [comment]
> > It needs to be rebased, apparently it doesn't apply clearly.
> 
> Do you have suggestions on how to go about that?

Get current sources, try to apply the patch you want, fix merge conflicts / rejected changes and make a diff from that.

**Comment by vulongm** [comment]
@grulja I was able to get the patch working but it seems to be running into a new issue. The folder selector does pop open successful however pressing ok or cancel results in the rest of the program freezing and being non-responsive.

**Comment by grulja** [comment]
> @grulja I was able to get the patch working but it seems to be running into a new issue. The folder selector does pop open successful however pressing ok or cancel results in the rest of the program freezing and being non-responsive.

Looks like https://bugs.kde.org/show_bug.cgi?id=438608 which I have never been able to reproduce.

**Comment by vulongm** [comment]
@grulja Not sure if it's quite the same bug. I am able to test out the folder selector in Kdenlive which seems to be fine there. Mostly only getting it on Krita so far.

**Comment by vulongm** [comment]
The bug does not seem to be occurring on Fedora Silverblue with GNOME testing it out there with the GNOME portal. Adding in the patch seems to cause the freezing issue to occur on both GNOME and KDE Plasma so might be better to leave it out for now to prevent work being lost when closing out the app.

For now the best workaround for this seems to be manually setting setting the directory in kritarc at ~/.var/app/org.kde.krita/config/kritarc.
```
recorder/snapshotdirectory=/var/home/user/KritaRecording
recorder_export/videodirectory=/var/home/user/Videos
```

**Comment by vulongm** [comment]
@travier Do you think it would be a good idea to disable flatpak portals or use it across different desktops? @TheEvilSkeleton found that on GNOME Krita uses the Qt file picker without a portal by default. That can be controlled using ``XDG_CURRENT_DESKTOP=KDE`` to turn on the portals there. But the portals also result in a workaround being required for this bug when on.

**Comment by travier** [comment]
Both options have downsides. Not sure which one is the best.

**Comment by vulongm** [comment]
Issue should be resolved now with #75 which was done by setting ``XDG_CURRENT_DESKTOP=gtk3`` resulting in a GTK non-portal directory selector.

**Comment by vulongm** [comment]
Reopening the issue as it occurs with the newer Krita Qt 5.15 fork in #177 even with the earlier workaround.
